qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -31,7 +33,49 @@
 )
 
 
-class MockRestlessBackend(FakeOpenPulse2Q):
+class FakeOpenPulse2QV2(FakeBackendV2):
```

Comment:
I'm curious about the reason not to use [BackendV2Converter](https://github.com/Qiskit/qiskit-terra/blob/a1a67a11741000f8f3b6d6a8766dab7e3cd06847/qiskit/providers/backend_compat.py#L184).

Reply:
The main reason is that I started this PR before it existed. I wonder if it would work or would suffer from the issues that I hack around below.

Reply:
I tried this:

```python
from qiskit.providers.fake_provider import FakeOpenPulse2Q
from qiskit.providers.backend_compat import BackendV2Converter

class FakeOpenPulse2QV2(BackendV2Converter):
    def __init__(self):
        v1backend = FakeOpenPulse2Q()
        super().__init__(v1backend)
```

but I encountered several test errors. A lot of them were this problem:

```
  File "/home/wshanks/Documents/Code/qiskit-experiments/test/library/characterization/test_half_angle.py", line 39, in test_end_to_end
    exp_data = hac.run(backend)                                                                                                                                
  File "/home/wshanks/Documents/Code/qiskit-experiments/qiskit_experiments/framework/base_experiment.py", line 238, in run
    transpiled_circuits = experiment._transpiled_circuits()                                                                                                    
  File "/home/wshanks/Documents/Code/qiskit-experiments/qiskit_experiments/framework/base_experiment.py", line 319, in _transpiled_circuits
    transpiled = transpile(self.circuits(), self.backend, **transpile_opts)                                                                                    
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/compiler/transpiler.py", line 326, in transpile
    unique_transpile_args, shared_args = _parse_transpile_args(                                                                                                
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/compiler/transpiler.py", line 656, in _parse_transpile_args                       
    inst_map = _parse_inst_map(inst_map, backend)                                                                                                              
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/compiler/transpiler.py", line 818, in _parse_inst_map                             
    inst_map = backend.target.instruction_schedule_map()                                                                                                       
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/providers/backend_compat.py", line 244, in target                                 
    self._target = convert_to_target(                                                                                                                          
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/providers/backend_compat.py", line 87, in convert_to_target                       
    duration=properties.readout_length(qubit),                                                                                                                 
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/providers/models/backendproperties.py", line 453, in readout_length
    return self.qubit_property(qubit, "readout_length")[0]  # Throw away datetime at index 1                                                    
  File "/home/wshanks/.conda/envs/qiskit/lib/python3.10/site-packages/qiskit/providers/models/backendproperties.py", line 389, in qubit_property
    raise BackendPropertyError(                                                                                                                                
qiskit.providers.exceptions.BackendPropertyError: "Couldn't find the property 'readout_length' for qubit 0." 
```

`FakeBackendV2` follows a different code path from `BackendConverter`. They use different `convert_to_target` functions. `BackendConverter` assumes that `readout_length` is in the properties here:

https://github.com/Qiskit/qiskit-terra/blob/a1a67a11741000f8f3b6d6a8766dab7e3cd06847/qiskit/providers/backend_compat.py#L86-L89

while `FakeBackendV2` checks for `readout_length` but does not error if it is not there:

https://github.com/Qiskit/qiskit-terra/blob/a1a67a11741000f8f3b6d6a8766dab7e3cd06847/qiskit/providers/fake_provider/utils/backend_converter.py#L80-L85

I tried working around that with this (thisi is also working around errors from `description` and `max_experiments` being missing):

```python
from qiskit.providers.fake_provider import FakeOpenPulse2Q
from qiskit.providers.backend_compat import BackendV2Converter


class FakeOpenPulse2QV2(BackendV2Converter):
    def __init__(self):
        backend_v1 = FakeOpenPulse2Q()
        backend_v1.configuration().description = "FakeOpenPulse2Q"
        backend_v1.configuration().max_experiments = 100
        cmd_def = backend_v1.defaults().cmd_def
        measure = next(c for c in cmd_def if c.name == "measure")
        readout_length = max(getattr(i, "duration", 0) for i in measure.sequence)
        for qubit in backend_v1._properties._qubits.values():
            qubit["readout_length"] = (readout_length * backend_v1.configuration().dt, qubit["readout_error"][1])
        super().__init__(backend_v1, add_delay=True)
        self._defaults = backend_v1._defaults
```

but I still had further errors. The next one that needed to be addressed was that `u2` and `id` were in the basis gates but not in the properties which led to an error in a transpiler pass.

At that point, I decided it was more trouble than it was worth to get this working. Maybe `BackendConverter` should be more robust, but likely `FakeOpenPulse2Q` just does not have complete enough metadata to match what is expected of a BackendV1 instance. We could fix that but likely it would be better just to make a BackendV2 equivalent class directly in terra.

Reply:
Thanks Will. This is good to know. I agree we should have Fake V2 in Terra in future.